src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,133 @@
 package core.basesyntax;
 
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final int DEFAULT_CAPACITY = 16;
+    private static final float LOAD_FACTOR = 0.75f;
 
-    @Override
-    public void put(K key, V value) {
+    private int size;
+    private Node<K, V>[] table;
+    private final float loadFactor;
+
+    public MyHashMap() {
+        this(DEFAULT_CAPACITY, LOAD_FACTOR);
+    }
+
+    public MyHashMap(int initialCapacity, float loadFactor) {
+        if (initialCapacity <= 0) {
+            throw new IllegalArgumentException("Initial capacity must be positive");
+        }
+        this.table = new Node[initialCapacity];
+        this.loadFactor = loadFactor;
+    }
 
+    public void put(K key, V value) {
+        if (key == null) {
+            putForNullKey(value);
+            return;
+        }
```

Comment:
Do not make separate logic for null key. It should be handled as any other one. Yes, it would be always in zero bucket but it doesn't change our approach